keyserver/listener.go

```diff
@@ -43,30 +43,25 @@ func (server *ConiksServer) listenForRequests(ln *net.TCPListener, handler func(
 
 func (server *ConiksServer) acceptClient(conn net.Conn, handler func(msg []byte) ([]byte, error)) {
 	defer conn.Close()
-
 	conn.SetDeadline(time.Now().Add(5 * time.Second))
+	tlsConn := tls.Server(conn, server.tlsConfig)
 
-	// handle request
-	buf := make([]byte, 4096)
-	for {
-		n, err := conn.Read(buf)
-		if err != nil {
-			if err != io.EOF {
-				log.Printf("client read %v: %v", conn.RemoteAddr(), err)
-			}
-			return
-		}
+	var buf bytes.Buffer
+	if _, err := io.CopyN(&buf, tlsConn, 8192); err != nil && err != io.EOF {
+		log.Printf("client read %v: %v", conn.RemoteAddr(), err)
+		return
+	}
 
-		res, err := handler(buf[:n])
-		if err != nil {
-			log.Printf("client handle %v: %v", conn.RemoteAddr(), err)
-		}
+	res, err := handler(buf.Bytes())
+	if err != nil {
+		log.Printf("client handle %v: %v", conn.RemoteAddr(), err)
+		// FIXME: what're we doing with the error here?
```

Comment:
I see.  `logMsg` would be clearer, yes.  Also, if the type was `error` that'd be good too.